chain/chain/src/chain.rs

```diff
@@ -1302,15 +1312,20 @@ impl Chain {
         }
         let mut missing = vec![];
         let block_height = block.header().height();
-        for (shard_id, chunk_header) in block.chunks().iter().enumerate() {
+
+        let epoch_id = block.header().epoch_id();
+        let shard_layout = self.epoch_manager.get_shard_layout(&epoch_id)?;
```

Comment:
Yeah I thought about it but it feels wrong having the epoch manager repeat the
shard layout API. I agree that it would help with readability. Let mef finish
this PR first and think about it.